Sources/GRPC/AsyncAwaitSupport/GRPCAsyncServerCallContext.swift

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if compiler(>=5.5)
+
+import Logging
+import NIOConcurrencyHelpers
+import NIOHPACK
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public final class GRPCAsyncServerCallContext {
```

Comment:
FWIW: I'm considering this "fine for now" rather than complete and signed off.

Reply:
@fabianfett I asked whether this was needed in the above resolved thread and apparently no. 
https://github.com/grpc/grpc-swift/pull/1249/files#r700021649

To avoid ping-ponging a fixup, I'll wait for you and @glbrntt to agree on this one.

Reply:
Any convergence between reviewers on this one?

Reply:
Ah, apologies, I missed this one.

---

@fabianfett It is fundamentally a value type but not one we want copy-on-write semantics with. The instance that the async handler holds must not diverge from the instance the implementor of the RPC holds. They hold these instances on different threads (EventLoop vs Task).

We could wrap this `class` up as a `struct` and pass it `inout` to the RPC. This would make the fact that the type stores mutable state more obvious. However without copy-on-write semantics this would be weird and surprising I think.

We also considered an `actor` but that felt clunky ("why do I have to `await` retrieving a logger or updating trailers?") and requires a `promise` each time the async server handler requires reading a property.

Reply:
Okay, can we get this as a `//` comment? I think this context is rather important when we look at it the next time.

Reply:
Immortalised this reasoning in a comment in 732f8648 😉